suggester/src/main/java/org/opengrok/suggest/popular/impl/chronicle/BytesRefSizedReader.java

```diff
@@ -46,6 +46,7 @@ private BytesRefSizedReader() {
 
     @NotNull
     @Override
+    @SuppressWarnings("rawtypes")
```

Comment:
I think this is a bogus from IDEA. We are building with `-Werror -Xlint:all` so if those do not report anything, then I'd not bother with it.

Reply:
it just blocks the compilation in that environment. Overriding the options for each branch I work on is tedious.

Reply:
Can you please be more specific what fails? 

If I add any line which causes warning such as `ArrayList<String> aa = new ArrayList(Arrays.asList("a"));`, then the compilation fails so `-Werror -Xlint:all` seem to work ok.

I've tried compiling with JDK8 and JDK15 and there seems to be no difference.

https://docs.oracle.com/javase/8/docs/api/java/lang/SuppressWarnings.html says that it should be used for compiler warnings.

If the compiler is not producing any warnings, then how come IDEA fails?

Reply:
Also, I think the warning is legitimate - `Bytes` is a raw type after all.

Reply:
> Also, I think the warning is legitimate - Bytes is a raw type after all.

What I'm concerned about is why `javac` is not reporting anything.

Reply:
If I checkout master at efa341d015c439bf6b7b31ebba695c025505d71e and compile the project (Ctrl+F9) in IDEA Ultimate 2020.2, it results in the error:
![Screenshot from 2020-10-06 16-20-04](https://user-images.githubusercontent.com/2934284/95214161-e70e8a00-07ef-11eb-99af-82092c0ce842.png)
The IDE is setup to use `/usr/lib/jvm/default-java/bin/java` which is currently:
```
$ /usr/lib/jvm/default-java/bin/java -version
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-2ubuntu219.10)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-2ubuntu219.10, mixed mode, sharing)
```

Reply:
Checking the 'Java compiler' settings it has these options for the 'suggester' module where this code resides:
![Screenshot from 2020-10-06 16-24-24](https://user-images.githubusercontent.com/2934284/95214748-83389100-07f0-11eb-8f07-b8e94dc16b2a.png)

Reply:
If you run `export JAVA_HOME=/usr/lib/jvm/default-java/ && ./mvnw clean compile`, then it works fine?

Reply:
And yes, I don't see this in the builds done via Maven.

Reply:
Checking the output in the 'Build Output' tab it indeed uses Java 11.0.7:
```
Executing pre-compile tasks...
Loading Ant Configuration...
Running Ant Tasks...
Running 'before' tasks
Checking sources
Copying resources... [suggester]
Copying resources... [plugins]
Copying resources... [opengrok-web]
Copying resources... [opengrok]
Parsing java... [suggester]
java: Errors occurred while compiling module 'suggester'
Checking dependencies... [suggester]
Dependency analysis found 0 affected files
javac 11.0.7 was used to compile java sources
Finished, saving caches...
Module "opengrok-dist" was fully rebuilt due to project configuration/dependencies changes
Compilation failed: errors: 1; warnings: 1
Executing post-compile tasks...
Loading Ant Configuration...
Running Ant Tasks...
Synchronizing output directories...
06.10.20 16:16 - Build completed with 1 error and 1 warning in 2 s 616 ms
```

Reply:
Hmm, this is weird. I'd say that this is IDEA bug. IMHO it should not report warnings not found by the compiler.

Reply:
Well, it is strange. I wonder how IDEA runs `javac`. Bug or not, it hampers my work :-)

Reply:
Changed the log4j configuration of IDEA per https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files and here's relevant excerpt from the `build.log` file:
```
2020-10-06 16:49:17,295 [   1829]  DEBUG - s.incremental.java.JavaBuilder - Compiling chunk [suggester] with options: "-g -deprecation -Xlint:all -Werror -Xlint:-auxiliaryclass -Xlint:-fallthrough -encoding UTF-
8 --release 8 -s /home/vkotal/opengrok-git-vladak/suggester/target/generated-sources/annotations", mode=in-process
2020-10-06 16:49:17,785 [   2319]  DEBUG - s.incremental.java.JavaBuilder - java:WARNING:found raw type: net.openhft.chronicle.bytes.Bytes
  missing type arguments for generic class net.openhft.chronicle.bytes.Bytes<Underlying>; file: /home/vkotal/opengrok-git-vladak/suggester/src/main/java/org/opengrok/suggest/popular/impl/chronicle/BytesRefSizedR
eader.java at (49:26)
2020-10-06 16:49:17,819 [   2353]  DEBUG - s.incremental.java.JavaBuilder - java:ERROR:warnings found and -Werror specified; file: /home/vkotal/opengrok-git-vladak/suggester/src/main/java/org/opengrok/suggest/po
pular/impl/chronicle/BytesRefSizedReader.java
2020-10-06 16:49:17,878 [   2412]  DEBUG - s.incremental.java.JavaBuilder - java:JPS_INFO:Errors occurred while compiling module 'suggester'
2020-10-06 16:49:17,894 [   2428]  DEBUG - s.java.dependencyView.Mappings - Begin of Differentiate:
2020-10-06 16:49:17,894 [   2428]  DEBUG - s.java.dependencyView.Mappings - Easy mode: false
2020-10-06 16:49:17,894 [   2428]  DEBUG - s.java.dependencyView.Mappings - End of Differentiate.
2020-10-06 16:49:17,894 [   2428]  DEBUG - brains.jps.incremental.Builder - Differentiate Results:
2020-10-06 16:49:17,894 [   2428]  DEBUG - brains.jps.incremental.Builder -    Compiled Files:
2020-10-06 16:49:17,894 [   2428]  DEBUG - brains.jps.incremental.Builder -    Affected Files:
2020-10-06 16:49:17,894 [   2428]  DEBUG - brains.jps.incremental.Builder -       /home/vkotal/opengrok-git-vladak/suggester/src/main/java/org/opengrok/suggest/popular/impl/chronicle/BytesRefSizedReader.java
2020-10-06 16:49:17,894 [   2428]  DEBUG - brains.jps.incremental.Builder - End Of Differentiate Results.
```